app/uk/gov/hmrc/tai/forms/EditIncomeForm.scala

```diff
@@ -106,7 +107,9 @@ object EditIncomeForm {
            errMessage: Option[String] = None)(implicit request: Request[_], messages: Messages) = createForm(taxablePayYTD, payDate, errMessage).bindFromRequest
 
   private def createForm (taxablePayYTD: BigDecimal, payDate: Option[LocalDate] = None, errMessage: Option[String] = None)(implicit messages: Messages) :Form[EditIncomeForm] = {
-    val date = if(payDate.isDefined) payDate.get.monthOfYear.getAsText() else ""
+
+    val date = payDate.map (date => DateHelper.monthOfYear(Dates.formatDate(date))).getOrElse("")
```

Comment:
Why are you calling the DateHelper to get the string value of the month? There should be a method on LocalDate to give you the month right?

Reply:
date.toString("MMMM") may give you what you want and that could be the function of monthOfYear method in your date helper instead of the regex check

Reply:
Happy with the discussion with @Gaur4vGaur